### PR TITLE
fix(cli): muti-typegraph bug + logging polish

### PR DESCRIPTION
**Pull request by Yohe-Am**
- Fixes multi-typegraph deployment bug.
- Fixes issue with loader process existing after cli death.
- Replaces `anyhow` with `color-eyre` in cli and improved logging story.
- `interlude` pattern was used to minimize changes by wrapping `eyre` through an `anyhow` export.
- [ ] Tests

MET-516
MET-517
MET-477

## Summary by CodeRabbit

- **Bug Fixes**
  - Corrected typos in configuration files to ensure proper authentication settings.

- **New Features**
  - Enhanced error handling with improved context and logging for better debugging.

- **Refactor**
  - Updated dependencies and imports across multiple modules for consistency and performance.
  - Reorganized error handling to use `eyre` library for more descriptive error messages.

- **Style**
  - Standardized logging and error message formats for clarity and uniformity.

- **Chores**
  - Updated platform compatibility in configuration files.
  - Removed deprecated dependencies and added new ones to improve functionality and maintainability.

### CODE REVIEW

**Comment by Yohe-Am** [comment]
Sample screenshots:

When lade errors out:
image

At the end of typegraph deployment if errors were seen:
image

File: meta-cli/src/typegraph/loader/mod.rs [hunk 1]
```diff
@@ -98,15 +93,24 @@ impl<'a> Loader<'a> {
...
             .env("META_CLI_TG_PATH", path.display().to_string())
-            .env("META_CLI_SERVER_PORT", get_instance_port().to_string())
+            .env("META_CLI_SERVER_PORT", get_instance_port().to_string());
+
+        use process_wrap::tokio::*;
+        let mut child = TokioCommandWrap::from(command)
+            .wrap(KillOnDrop)
+            // we use sessions so that kill on drop
+            // signals will get all grand-children
+            .wrap(ProcessSession)
```
Comment (afmika) [inline meta-cli/src/typegraph/loader/mod.rs:113]: Nice!

File: examples/metatype.yaml [hunk 2]
```diff
@@ -4,9 +4,9 @@ typegates:
     username: admin
     password: password
     secrets:
-      basic-authentification:
-        BASIC_ADMIN: "password"
-      jwt-authentification:
+      basic-authentication:
+        BASIC_admin: "password"
```
[the remark this answers is not in the document]
Comment (Yohe-Am) [reply examples/metatype.yaml:8]: These are sample configurations as one can see by the path of the file.

File: meta-cli/src/typegraph/loader/mod.rs [hunk 3]
```diff
@@ -78,17 +72,18 @@ impl LoaderPool {
...
                 if !exists {
                     return Err(LoaderError::ModuleFileNotFound { path });
                 }
             }
-            Err(e) => {
+            Err(err) => {
                 return Err(LoaderError::Unknown {
                     path,
-                    error: anyhow!("failed to check if file exists: {}", e.to_string()),
+                    error: ferr!("failed to check if file exists: {err:#}",),
```
[the remark this answers is not in the document]
Comment (Yohe-Am) [reply meta-cli/src/typegraph/loader/mod.rs:86]: Error messages shouldn't be capitalized to maximize composition.

File: meta-cli/src/typegraph/loader/mod.rs [hunk 4]
```diff
@@ -173,15 +177,19 @@
                         None => "".to_string(),
                     };
 
+                    use color_eyre::SectionExt;
                     Err(LoaderError::LoaderProcess {
                         path: path.clone(),
-                        error: anyhow::anyhow!("{}\n{}", stderr, stdout),
+                        error: ferr!("loader process err")
+                            .with_section(move || stdout.trim().to_string().header("Stdout:"))
+                            .with_section(move || stderr.trim().to_string().header("Stderr:")),
```
[the remark this answers is not in the document]
Comment (Yohe-Am) [reply meta-cli/src/typegraph/loader/mod.rs:185]: This isn't a log message but an error value.

**Review by afmika — APPROVED** [review]

**Review by afmika — APPROVED** [review]
lgtm

**Comment by Yohe-Am** [comment]
@coderabbitai based on the patterns established in typegate/tests/e2e/cli/dev_test.ts and typegate/tests/e2e/website/website_test.ts, generate a new test file inside typegate/tests/e2e/cli/ that directory that has tests for deployment of multiple typegraphs at once. Name a few typegraphs from the examples directory and specify them using the `-f` flag.

**Pull request merged**